azfile/zc_credential_token_test.go

```diff
@@ -0,0 +1,47 @@
+package azfile_test
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/aymanjarrousms/azure-storage-file-go/azfile"
+
+	chk "gopkg.in/check.v1"
+)
+
+type OAuthSuite struct{}
+
+var _ = chk.Suite(&OAuthSuite{})
+
+func getShareName() string {
+	name := os.Getenv("SHARE_NAME")
+
+	if name == "" {
+		panic("SHARE_NAME environment vars must be set before running oauth tests")
+	}
+
+	return name
+}
+
+func (s *OAuthSuite) TestFileShareOAuth(c *chk.C) {
+	fsu := getFSUWithOAuth()
+
+	shareURL := fsu.NewShareURL(getShareName())
```

Comment:
Please create and delete the file share using a fsuWithAccountKey. see suggestion below.
Maybe add a comment that the storage account name and key for this test to pass must be in a Canary region.

```suggestion
	fsuWithAccountKey := getFSU()
	shareURLWithAccountKey, shareName := createNewShare(c, fsuWithAccountKey)
	defer delShare(c, shareURLWithAccountKey, azfile.DeleteSnapshotsOptionNone)

	fsu := getFSUWithOAuth()

	shareURL := fsu.NewShareURL(shareName)
```